### PR TITLE
supporting dump command flag for MariaDB

### LINKED ISSUE #370: BinlogSyncer.writeBinlogDumpCommand flag is hardcoded and only supports mysql
MySQL 和 MariaDB  对`COM_BINLOG_DUMP ` 中的 flag 有不同的值，目前我们写死为0。对于需要 `BINLOG_SEND_ANNOTATE_ROWS_EVENT ` 的场景无法满足。

其实应该搞成可以配置的。

### PR DESCRIPTION
**Pull request by imjustfly**
try to fix #370 by making binlog dump command's flag configurable.

### CODE REVIEW

**Comment by GregoryIan** [comment]
please update to a more appropriate name

**Comment by GregoryIan** [comment]
thanks your PR @imjustfly , can you add some test cases, perhaps in `replication_test.go`

**Comment by imjustfly** [comment]
@GregoryIan 

Emm.. You mean `DumpFlag` ? What about `DumpCommandFlag` or `BinlogDumpCommandFlag` ? What's your opinion?

And OK,  I try to add some test later.

File: go.mod [hunk 1]
```diff
@@ -2,9 +2,15 @@ module github.com/siddontang/go-mysql
 
 require (
 	github.com/BurntSushi/toml v0.3.1
+	github.com/go-sql-driver/mysql v1.4.1
```
Comment (siddontang) [inline go.mod]: em, why do you introduce these dependencies, seem they are not relevant here.

**Comment by siddontang** [comment]
@imjustfly 

I don't think you need to change mod in this PR

File: go.mod (hunk 1 above)
Comment (imjustfly) [reply go.mod]: these dependencies are introduced after running `go mod tidy`.

for example:

```
go mod why -m github.com/go-sql-driver/mysql
# github.com/go-sql-driver/mysql
github.com/siddontang/go-mysql/server
github.com/siddontang/go-mysql/server.test
github.com/go-sql-driver/mysql
```

`github.com/go-sql-driver/mysql` is introduce at https://github.com/siddontang/go-mysql/blob/7263f5610b7a05bfb0ceff83e99ada9f5287a904/server/server_test.go#L13

**Comment by imjustfly** [comment]
I changed the name of config item and added a simple test according to existing test suite. PTAL @GregoryIan @siddontang

File: replication/replication_test.go [hunk 2]
```diff
@@ -369,6 +369,12 @@ func (t *testSyncerSuite) TestMariadbGTIDSync(c *C) {
 	t.testSync(c, s)
 }
 
+func (t *testSyncerSuite) TestMariadbAnnotateRows(c *C) {
```
Comment (GregoryIan) [inline replication/replication_test.go]: 😎 now we have testing case, but I think it's cooler to confirm that we can receive `annotate rows event`  if enabling in test casing.  What you do think about it @imjustfly

Comment (imjustfly) [reply replication/replication_test.go]: Current testing logic is not scalable enough, all testing cases are in `testSync`. And it just ensure there are no errors in event parsing and doesn't check the event type and content. 

If you want to do so, I think it's better to refactor the testing logic and added this event checking at that time.

Comment (GregoryIan) [reply replication/replication_test.go]: would you like to do it？

**Comment by GregoryIan** [comment]
LGTM now

File: replication/replication_test.go (hunk 2 above)
Comment (imjustfly) [reply replication/replication_test.go]: To be honest, not now. 

I am busy working on syncing data between mariadb now. I just want to fix this issue quickly so I don't need to maintain a `go-mysql` repo by myself. And I think this PR is acceptable at the current existing testing level.  

Refactoring testing should be done by anyone who cares about it, including me, in the future.

Comment (GregoryIan) [reply replication/replication_test.go]: LOL, feel free

**Pull request merged**